### PR TITLE
Add other/restrict-ingress-defaultbackend policy 

### PR DESCRIPTION
**Pull request by aslafy-z**
## Related Issue(s)

Fixes https://github.com/kyverno/policies/issues/322

## Description

Create a sample policy to prevent usage of the Ingress defaultBackend field.

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [x] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.

### CODE REVIEW

File: other/restrict_ingress_defaultbackend/restrict_ingress_defaultbackend.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      objects, the traffic is routed to your default backend. In a multi-tenant environment, you
+      want users to use explicit hosts, they should not be able to overwrite the global default backend
+      service.
+spec:
+  validationFailureAction: audit
+  background: true
+  rules:
+  - name: restrict-ingress-defaultbackend
+    match:
+      resources:
```
Comment (chipzoller) [inline other/restrict_ingress_defaultbackend/restrict_ingress_defaultbackend.yaml:22]: Please place this under an `any` block as Kyverno will be moving away from non-`any` | `all` blocks in `match` or `exclude` clauses.

**Comment by chipzoller** [comment]
Thank you for this contribution! Do try and look at the [contribution guidelines](https://github.com/kyverno/policies#contribution) and especially complete the annotations section as these are what allow the user-friendly filtering on the [website](https://kyverno.io/policies/).

**Review by chipzoller — CHANGES_REQUESTED** [review]
Minor clarifications.

File: other/restrict_ingress_defaultbackend/restrict_ingress_defaultbackend.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    policies.kyverno.io/minversion: 1.6.0
+    kyverno.io/kubernetes-version: "1.19-1.24"
+    policies.kyverno.io/subject: Ingress
+    policies.kyverno.io/description: >-
+      An Ingress with no rules sends all traffic to a single default backend. The defaultBackend
+      is conventionally a configuration option of the Ingress controller and is not specified in
+      your Ingress resources. If none of the hosts or paths match the HTTP request in the Ingress
+      objects, the traffic is routed to your default backend. In a multi-tenant environment, you
+      want users to use explicit hosts, they should not be able to overwrite the global default backend
+      service.
```
Comment (aslafy-z) [inline other/restrict_ingress_defaultbackend/restrict_ingress_defaultbackend.yaml:19]:
```suggestion
      service. This policy prohibits the use of the defaultBackend field.
```

File: other/restrict_ingress_defaultbackend/restrict_ingress_defaultbackend.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: restrict-ingress-defaultbackend
+  annotations:
+    policies.kyverno.io/title: Restrict Ingress defaultBackend
+    policies.kyverno.io/category: Best Practices
+    policies.kyverno.io/severity: high
+    kyverno.io/kyverno-version: 1.6.0
```
[the remark this answers is not in the document]
Comment (aslafy-z) [reply other/restrict_ingress_defaultbackend/restrict_ingress_defaultbackend.yaml:9]: I did not test the policy from my end at the time.

File: other/restrict_ingress_defaultbackend/restrict_ingress_defaultbackend.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    policies.kyverno.io/minversion: 1.6.0
+    kyverno.io/kubernetes-version: "1.19-1.24"
```
[the remark this answers is not in the document]
Comment (aslafy-z) [reply other/restrict_ingress_defaultbackend/restrict_ingress_defaultbackend.yaml:11]: I did not test the policy from my end at the time.

File: other/restrict_ingress_defaultbackend/restrict_ingress_defaultbackend.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      want users to use explicit hosts, they should not be able to overwrite the global default backend
+      service. This policy prohibits the use of the defaultBackend field.
+spec:
+  validationFailureAction: audit
+  background: true
+  rules:
+  - name: restrict-ingress-defaultbackend
+    match:
+      any:
+        resources:
```
Comment (aslafy-z) [inline other/restrict_ingress_defaultbackend/restrict_ingress_defaultbackend.yaml:27]:
```suggestion
      - resources:
```

**Review by chipzoller — APPROVED** [review]
Thank you! I'll have a test and update the annotations accordingly.

**Pull request merged**